### ISSUE TITLE
On-demand avatar fetching

### ISSUE THREAD

**Issue by kofalt** [scitran/core#192]
Right now, we need internet connectivity and the gravitar service to add a user to the system. This was originally a way to get around some multi-avatar-provider design problems, but I've sat for enough seconds at `INFO bootstrapping users...` to get fed up with it :smile: 

Adding a user should not require a 3rd-party service - or at least, not for this. It slows down setup & admin user creation, and you may not need that information for a long time... if at all.

How about fetching it on-demand? An `<img>` tag is the perfect place to stash this latency, because it doesn't block anything. We _could_ start storing/caching the avatars themselves, but I don't think we need to tackle that right now.

Instead, I propose we `307 Temporary Redirect` everyone on a new route, possibly `api/users/id/avatar`, that returns a redirect to the correct URL. And this route does whatever avatar-choosing setup work as needed.

**Comment by gsfr** [comment]
Overkill. Just use a shorter bootstrap file when testing on your own.

**Comment by kofalt** [comment]
This ticket could have been stated differently, because I ended up tackling a few more important things:

* `RequestHandler.__init__` has historically been the most cyclomaticly-complex function we have; it's now much simpler and organized.
* Google- and Gravatar-specific logic are now encapsulated in their own functions, rather than intertwined with other stuff.
* The new route proposal was well-received by the web guys, so things like displaying an avatar on each comment, for example, will be much simpler.
* Logic is now more-ready to be broken out from stateful classes.

So basically a few contributions to the code review from late last year.
Cleaning up the connectivity problem is just a side benefit :)

I've got a PR ready, but for convenience I based it on the `origin` branch.
When that lands have a look, it's very simple - mostly cut & paste.
